File: src/LMS_Assignment.Domain/Entities/TeacherSubjectAssignment.cs
namespace LMS_Assignment.Domain.Entities;

public class TeacherSubjectAssignment
{
    public Guid Id { get; set; }
    public Guid TeacherId { get; set; }
    public Guid ClassSubjectId { get; set; }
    public Guid? AssignedBy { get; set; }
    public DateTime AssignedAt { get; set; } = DateTime.UtcNow;

    // Navigation properties
    public User Teacher { get; set; } = null!;
    public ClassSubject ClassSubject { get; set; } = null!;
    public User? AssignedByUser { get; set; }
    public ICollection<Assignment> Assignments { get; set; } = new List<Assignment>();
}

File: frontend/src/types/teacher-subject-assignment.ts
export interface TeacherSubjectAssignment {
  id: string
  teacherId: string
  teacherName: string
  classSubjectId: string
  className: string
  subjectName: string
  assignedBy: string
  assignedAt: string
}

export interface CreateTeacherSubjectAssignmentRequest {
  teacherId: string
  classSubjectId: string
}
